ui/src/overridden/frontsite/Document/DocumentDetails/DocumentMetadataTabs/Identifiers/Identifiers.js

```diff
@@ -0,0 +1,154 @@
+import {
+  InfoPopup,
+  SeparatedList,
+} from '@inveniosoftware/react-invenio-app-ils';
+import capitalize from 'lodash/capitalize';
+import _isEmpty from 'lodash/isEmpty';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Divider, Table, Loader } from 'semantic-ui-react';
+
+// component not to be used in search pages
+// might trigger a lot of requests
+export class Identifiers extends Component {
+  componentDidMount() {
+    const { vocabularyIdentifiers, fetchVocabularyIdentifiers } = this.props;
+    if (!vocabularyIdentifiers) {
```

Comment:
I am sorry to stress on this, but here we are mixing responsibilities.
It is not a responsibility of a UI component to check for the existence of data and then do something: this code means that a UI component knows when and how an HTTP request has to be made. It also knows that `fetchVocabularyIdentifiers` is performing an HTTP call.
This should happen in the business logic layer:
* if you want to reuse `fetchVocabularyIdentifiers` in another cmp, you will have to reimplement the same `if`, over and over
* testing is harder

To be very clear: the UI should never know how the data is fetched.

Please refactor the code, here the pseudocode. 

Each specific vocab cmp should do:

```
componentDidMount(..)
  fetchVocabularyScheme(identifierScheme)
```

The Redux action, your business logic should hide the caching part to the external world:

```
  function fetchVocabularyScheme(identifierScheme)...
      identifierSchemeValue = mycache.getVocabularyScheme(identifierScheme)
      if not identifierSchemeValue:
          identifierSchemeValue = fetchFromBackend(identifierScheme)
          mycache.setVocabularyScheme(identifierSchemeValue)
      
      setState(... identifierSchemeValue ...)
```

Let's discuss this at the stand up.